external/fv3fit/fv3fit/reservoir/readout.py

```diff
@@ -74,11 +78,23 @@ class ReservoirComputingReadout:
     _INTERCEPTS_NAME = "intercepts.npy"
 
     def __init__(self, coefficients: np.ndarray, intercepts: np.ndarray):
+        # coefficients: [(subdomain), in_features, out_features]
+        # intercepts: [(subdomain), out_features]
         self.coefficients = coefficients
         self.intercepts = intercepts
 
+        if len(coefficients.shape) == 2:
```

Comment:
Efficient way to handle matrix multiplication with arbitrary leading dimensions.  Pinned the coefficients to be a specific number of dimensions.

Always provides the sum product over the last dimension of the input with the 2nd to last dimension of the outputs.